cumulusci/tasks/salesforce/EnsureRecordTypes.py

```diff
@@ -0,0 +1,140 @@
+            <default>false</default>
+        </values>
+    </businessProcesses>"""
+
+BUSINESS_PROCESS_LINK = (
+    """<businessProcess>{record_type_developer_name}</businessProcess>"""
+)
+
+SOBJECT_METADATA = """<?xml version="1.0" encoding="utf-8"?>
+<CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+    {business_process_metadata}
+    <recordTypes>
+        <fullName>{record_type_developer_name}</fullName>
+        <active>true</active>
+        {business_process_link}
+        <label>{record_type_label}</label>
+    </recordTypes>
+</CustomObject>
+"""
+
+PACKAGE_XML = """<?xml version="1.0" encoding="UTF-8"?>
+<Package xmlns="http://soap.sforce.com/2006/04/metadata">
+    <types>
+        <members>*</members>
+        <name>CustomObject</name>
+    </types>
+    <version>45.0</version>
+</Package>"""
+
+SOBJECT_MAP = {
+    "Opportunity": "StageName",
+    "Lead": "Status",
+    "Case": "Status",
+    "Solution": "Status",
+}
+
+
+class EnsureRecordTypes(BaseSalesforceApiTask):
+    task_options = {
+        "record_type_developer_name": {
+            "description": "The Developer Name of the Record Type (unique).  Must contain only alphanumeric characters and underscores.",
+            "required": True,
+        },
+        "record_type_label": {
+            "description": "The Label of the Record Type.",
+            "required": True,
+        },
+        "sobject": {
+            "description": "The sObject on which to deploy the Record Type and optional Business Process.",
+            "required": True,
+        },
+    }
+    _deploy = Deploy
+
+    def _init_options(self, kwargs):
+        super(EnsureRecordTypes, self)._init_options(kwargs)
+
+        self.options["generate_business_process"] = False
+
+        # Validate developer name
+        if not re.match(r"^\w+$", self.options["record_type_developer_name"]):
+            raise TaskOptionsError(
+                "Record Type Developer Name value must contain only alphanumeric or underscore characters"
+            )
+
+    def _infer_business_process(self):
+        # If our sObject is Lead or Opportunity, we need to generate businessProcess
+        # metadata to make the record type deployable.
+        sobject = self.options["sobject"]
+
+        if sobject in SOBJECT_MAP:
+            self.options["generate_business_process"] = True
+            describe_results = getattr(self.sf, sobject).describe()
+            # Salesforce requires that at least one picklist value be present and active
+            self.options["stage_name"] = list(
+                filter(
+                    lambda pl: pl["active"],
+                    list(
+                        filter(
+                            lambda f: f["name"] == SOBJECT_MAP[sobject],
+                            describe_results["fields"],
+                        )
+                    )[0]["picklistValues"],
+                )
+            )[0]["value"]
```

Comment:
This is mostly a matter of taste but I'd like to suggest that the following is more readable:

    stage_field = [f for f in describe_results["fields"] if f["name"] == SOBJECT_MAP[sobject]][0]
    active_picklist_values = [value for value in stage_field['picklistValues'] if value["active"]]
    self.options["stage_name"] = active_picklist_values[0]["value"]